cutil/logger/logger.go

```diff
@@ -137,6 +125,20 @@ func Warning(format string, a ...interface{}) {
 	}
 }
 
+func extractLoggerArgs(format string, a ...interface{}) ([]interface{}, io.Writer) {
```

Comment:
I am going to lean more towards @prateekgogia's suggestion of modeling our functions after `fmt.Fprint()` and explicitly calling out the `io.Writer` as a nilable argument to the function.

Another approach would be to have functions like: `log.Finfo` AND `log.Info` 

However you want to do it is fine with me, but I am worried about introducing a new paradigm that the last argument *should* be a certain type. If it's a type let's call it what it is right?